### PR TITLE
[core] Add Span.set_tag manual.keep and manual.drop support

### PR DESCRIPTION
**Pull request by brettlangdon**
This PR adds support for `Span.set_tag('force.keep')` which is an alias for setting the sampling priority to USER_KEEP.

### CODE REVIEW

**Comment by delner** [comment]
Why do this via a tag? Why not access the context via `span.context` and set it that way?

File: lib/ddtrace/span.rb [hunk 1]
```diff
@@ -68,7 +69,15 @@ def initialize(tracer, name, options = {})
     # must be strings. A valid example is:
     #
     #   span.set_tag('http.method', request.method)
-    def set_tag(key, value)
+    def set_tag(key, value = nil)
+      # 'force.keep' is an alias for setting the sampling priority to USER_KEEP
+      if key == 'force.keep'
```
Comment (palazzem) [inline lib/ddtrace/span.rb]: It must be `manual.keep` and `manual.drop`.

**Comment by delner** [comment]
Changed this so that it doesn't actually set a tag (just changes the trace context.) Also tried to fix 1.9.3 compatibility... ended up putting in a super-hack which is very brittle and likely to break if we change `Datadog::Span` with other modules. It would be tres bien if we could drop 1.9.3 support before then...

File: lib/ddtrace/span.rb [hunk 2]
```diff
@@ -256,3 +258,4 @@ def now_allocations
 
 # Include extensions after Span (for Ruby 1.9 compatibility)
 Datadog::Span.send(:include, Datadog::Analytics::Span)
+Datadog::Span.send(:include, Datadog::ForcedTracing::Span)
```
Comment (brettlangdon) [inline lib/ddtrace/span.rb]: We might want a note here about the importance of the ordering

**Pull request merged**